### PR TITLE
fix(neighbors): honor per-axis pbc when wrapping naive neighbor positions

### LINKED ISSUE #104: 🐛[BUG]: `batch_naive_neighbor_list` returns wrong neighbors for non-periodic axes (pbc=False) when a cell is provided, position wrapping ignores `pbc`
### Version

0.3.1 (and also latest main branch)

### On which installation method(s) does this occur?

Pip

## Summary

`batch_naive_neighbor_list` returns an incorrect neighbor list whenever a **non-periodic axis** is combined with a **non-zero cell** and atoms cross that axis' cell boundary.
Note that `batch_cell_list` returns correct values.

This affects

- fully non-periodic systems `pbc=[False, False, False]` (e.g. molecules)
- slabs `pbc=[True, True, False]` (the non-periodic z axis)

This is distinct from #18 (unwrapped periodic).

## Why it happens

A non-periodic axis has no images, so its neighbor list must be cell-independent with zero shifts.
The naive path breaks this:

1. `batch_naive_neighbor_list` branches only on `pbc is None`, so a fully non-periodic `pbc=[F,F,F]` still takes the **PBC** path (`_batch_naive_neighbor_matrix_pbc`) instead of the `_batch_naive_neighbor_matrix_no_pbc` kernel that would skip wrapping.
(A slab `pbc=[T,T,F]` *should* use the PBC path for x,y — it just must not wrap z.)
2. In that path the wrap kernel **`_wrap_positions_batch_kernel`** (`nvalchemiops/neighbors/neighbor_utils.py`) takes **no `pbc` argument** and folds *every* axis (`_int = floor(pos · inv_cell)`), while `_compute_naive_num_shifts` *does* honor `pbc` and disables image search on non-periodic axes.
3. As a result, every non-periodic axis is corrupted. Depending on the cell, the folded pair is either kept with a wrong shift (same count, wrong geometry — the molecule example below) or pushed beyond the cutoff and never recovered (fewer pairs — the slab example below).

Confirmation: passing `wrap_positions=False` to `batch_naive_neighbor_list` makes both cases correct.

### Trigger condition
Any non-zero cell with `pbc[d] = False` on at least one axis, plus atoms not all confined to `[0, cell_d)` on that axis.
This is the default for non-PBC use: `ase.build.molecule()` centers on the origin, XYZ/SDF files don't enforce a cell, and slab/2D MD drifts atoms across the non-periodic boundary over time.

## Suggested fix

Make the position wrapping in the naive PBC path `pbc`-aware so non-periodic axes are not folded. `_compute_naive_num_shifts` in the same file already follows this pattern (accepts `pbc`, zeroes the shift range on non-periodic axes); the wrap kernels should do the analogous thing.

### Minimum reproducible example

```python
import torch
from ase import Atoms
from ase.build import molecule
from ase.neighborlist import neighbor_list
from nvalchemiops.torch.neighbors import batch_cell_list, batch_naive_neighbor_list

device = "cuda" if torch.cuda.is_available() else "cpu"
dtype = torch.float64
cutoff = 5.0


def nvalchemiops_distances(fn, atoms):
    # Run the neighbor list and reconstruct the distance of every returned pair
    # from the ORIGINAL positions: |pos_j + shift @ cell - pos_i|. Comparing
    # distances (not just the pair count) also catches the case where the count
    # is right but the periodic `shift`s are wrong.
    positions = torch.tensor(atoms.get_positions(), dtype=dtype, device=device)
    cell = torch.tensor(atoms.cell.array, dtype=dtype, device=device)
    pbc = torch.tensor(atoms.pbc, device=device)
    batch_idx = torch.zeros(len(atoms), dtype=torch.int32, device=device)
    nbr, _, shifts = fn(
        positions=positions,
        cutoff=cutoff,
        batch_idx=batch_idx,
        cell=cell.unsqueeze(0),
        pbc=pbc.unsqueeze(0),
        return_neighbor_list=True,
    )
    i, j = nbr[0].long(), nbr[1].long()
    disp = positions[j] + shifts.to(dtype) @ cell - positions[i]
    return torch.sort(disp.norm(dim=1)).values.cpu()


def report(label, atoms):
    # ASE's neighbor_list is the trusted reference (honors per-axis pbc).
    ref = torch.sort(torch.tensor(neighbor_list("d", atoms, cutoff), dtype=dtype)).values

    def line(name, fn):
        d = nvalchemiops_distances(fn, atoms)
        ok = d.shape == ref.shape and torch.allclose(d, ref, atol=1e-6)
        return f"  {name:29s} = {d.shape[0]:4d} pairs   [{'OK' if ok else 'WRONG'}]"

    print(label)
    print(f"  {'ase.neighbor_list (reference)':29s} = {ref.shape[0]:4d} pairs")
    print(line("batch_naive_neighbor_list", batch_naive_neighbor_list))
    print(line("batch_cell_list", batch_cell_list))
    print()


# Case 1: non-periodic molecule (pbc=[F,F,F]) carrying a 1 A dummy cell.
# The cell is physically meaningless for a non-periodic system, but clients
# attach one (e.g. ase get_cell(complete=True) yields a 1 A identity).
mol = molecule("H2O")
mol.set_cell([1.0, 1.0, 1.0])  # dummy cell: non-zero, but pbc stays False
report("non-periodic molecule  pbc=[F,F,F], 1A dummy cell", mol)

# Case 2: slab, periodic in x,y and non-periodic in z (pbc=[T,T,F]),
# with atoms straddling the z=0 cell boundary.
a, Lz = 3.0, 20.0
slab = Atoms(
    "H8",
    positions=[[x, y, z] for (x, y) in [(0, 0), (1.5, 0), (0, 1.5), (1.5, 1.5)] for z in (0.5, -0.5)],
    cell=[a, a, Lz],
    pbc=[True, True, False],
)
report("slab  pbc=[T,T,F], z straddles 0", slab)
```

### Relevant log output

```shell
Output of the above code


non-periodic molecule  pbc=[F,F,F], 1A dummy cell
  ase.neighbor_list (reference) =    6 pairs
  batch_naive_neighbor_list     =    6 pairs   [WRONG]
  batch_cell_list               =    6 pairs   [OK]

slab  pbc=[T,T,F], z straddles 0
  ase.neighbor_list (reference) =  584 pairs
  batch_naive_neighbor_list     =  288 pairs   [WRONG]
  batch_cell_list               =  584 pairs   [OK]
```

### Environment details

```shell
- Python 3.12
- nvalchemi-toolkit-ops 0.3.1
- torch 2.9.0, warp 1.14.0, CUDA 12.8
```

### PR DESCRIPTION
**Pull request by gaoflow**
## Description

`batch_naive_neighbor_list` and `naive_neighbor_list` returned wrong neighbors whenever a non-periodic axis was combined with a non-zero cell and atoms crossed that axis' boundary. This affects fully non-periodic systems (`pbc=[False, False, False]`, for example a molecule carrying a dummy cell) and slabs (`pbc=[True, True, False]`).

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

## Related Issues

Fixes #104

## Changes Made

- `get_wrap_positions_kernel` now builds a `pbc_aware` variant of the position-wrapping kernel that leaves non-periodic axes unfolded, mirroring how `_compute_naive_num_shifts` already disables image search on those axes.
- The batch and single-system Torch neighbor lists thread their `pbc` through `_prepare_pbc_positions` to that variant. Selection is opt-in: the default kernel and any caller that does not pass `pbc` keep the previous fold-every-axis behavior, so the change stays confined to the periodic wrapping path. The dual-cutoff and JAX wrapping paths are untouched and can adopt the same `pbc` argument in a follow-up.
- Added a regression test and a CHANGELOG entry.

## Testing

- [ ] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

The new `test_non_periodic_axis_is_not_folded` reconstructs neighbor distances from the returned shifts and asserts the naive neighbor list matches `batch_cell_list` for a non-periodic molecule and a slab whose atoms straddle the non-periodic boundary; reverting the fix makes it fail. Both cases were also checked against ase `neighbor_list`. The full suite needs a GPU, so locally I ran the CPU-runnable neighbor tests (the new test plus the existing factory and batch-naive launcher tests) along with `ruff check`, `ruff format`, and `interrogate`; please rely on CI for the GPU paths.

## Checklist

- [x] I have read and understand the Contributing Guidelines
- [x] I have updated the CHANGELOG.md
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

Root cause: the naive launchers branch only on whether `pbc` is `None`, so a non-`None` but non-periodic `pbc` still takes the periodic path, where the wrapping kernel folded every axis with `floor(pos . inv_cell)`. A non-periodic axis has no images, so folding it changes the geometry and corrupts its neighbors. `batch_cell_list` was unaffected, which is why it is the reference in the new test.

### CODE REVIEW

**Comment by zubatyuk** [comment]
Thank you for the contribution. We were aware of this bug and prepared a more complete PR to fold into our 0.4.0 RC branch: https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/106
